tips/TIP-0018/tip-0018.md

```diff
@@ -0,0 +1,3755 @@
+                        </blockquote>
+                        <table>
+                            <tr>
+                                <td><b>Name</b></td>
+                                <td><b>Type</b></td>
+                                <td><b>Description</b></td>
+                            </tr>
+                            <tr>
+                                <td>Block Type</td>
+                                <td>uint8</td>
+                                <td>
+                                    Set to <strong>value 3</strong> to denote a <i>Tag Block</i>.
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Tag Length</td>
+                                <td>uint8</td>
+                                <td>
+                                    Length of the following tag field in bytes.
+                                </td>
+                            </tr>
+                            <tr>
+                                <td>Tag</td>
+                                <td>ByteArray</td>
+                                <td>Binary indexation data.</td>
+                            </tr>
+                        </table>
+                    </details>
+                </td>
+            </tr>
+        </table>
+    </details>
+</table>
+
+### Additional Transaction Syntactic Validation Rules
+
+- `Amount` field must fulfill the dust protection requirements.
+- `Native Tokens Count` must not be greater than `Max Native Tokens Count`.
+- `Native Tokens` must be lexicographically sorted based on `Token ID`.
+- Each <i>Native Token</i> must be unique in the set of `Native Tokens` based on its `Token ID`. No duplicates are
+  allowed.
+- It must hold true that `1` ≤ `Unlock Conditions Count` ≤ `4`.
+- `Unlock Condition Type` of an <i>Unlock Condition</i> must define on of the following types:
+  - <i>Address Unlock Condition</i>
+  - <i>Dust Deposit Return Unlock Condition</i>
+  - <i>Timelock Unlock Condition</i>
+  - <i>Expiration Unlock Condition</i>
+- <i>Unlock Conditions</i> must be sorted in ascending order based on their `Unlock Condition Type`.
+- Syntactic validation of all present unlock conditions must pass.
+- <i>Address Unlock Condition</i> must be present.
+- It must hold true that `0` ≤ `Blocks Count` ≤ `3`.
+- `Block Type` of a <i>Block</i> must define on of the following types:
+  - <i>Sender Block</i>
+  - <i>Metadata Block</i>
+  - <i>Tag Block</i>
+- <i>Blocks</i> must be sorted in ascending order based on their `Block Type`.
+- Syntactic validation of all present feature blocks must pass.
+
+### Additional Transaction Semantic Validation Rules
+
+#### Consumed Outputs
+
+- The unlock block of the input must correspond to `Address` field in the <i>Address Unlock Condition</i> and the
+  unlock must be valid.
+- The unlock is valid if and only if all unlock conditions and feature blocks present in the output validate.
```

Comment:
Is there actually anything that needs validation regarding feature blocks of consumed outputs ?

Reply:
Yes, there is one, the issuer block: you need to validate that it will be carried over to the next state of the UTXO state machine.

Reply:
Thanks